File: Linked_Lists/python/kth.py
from LinkedList import  LinkedList, Node
from random import randint


def k_to_last(head: Node, k):
    if head is None:
        return 0

    index  = k_to_last(head.next, k) + 1

    if index == k:
        print(k, head)

    return index


def k_to_last2(head: Node, k):
    p1 = head
    p2 = head

    for i in range(k):
        p2 = p2.next

    while p2.next:
        p1 = p1.next
        p2 = p2.next

    print(k, p1)


def main():
    linked_list = LinkedList()
    for i in range(10):
        linked_list.append(randint(0, 5))
    print(linked_list)

    k_to_last2(linked_list.head, 5)


if __name__ == '__main__':
    main()

File: Linked_Lists/go/kth.go
package main

import (
  "types"
  "math/rand"
  "fmt"
)


func kToLast(head *types.Node, k int) int {
  if head.Next == nil{
    return 0
  }

  index := kToLast(head.Next, k) + 1

  if index == k {
    fmt.Println(k, head.Data)
  }
  return index
}


func kToLast2(head *types.Node, k int) {
  p1 := head
  p2 := head

  for i := 0; i < k; i++ {
    p2 = p2.Next
  }

  for p2.Next != nil {
    p1 = p1.Next
    p2 = p2.Next
  }

  fmt.Println(k, p1.Data)

}


func main() {
  var list types.LinkedList
  var length int = 5

  for i := 0; i < length; i++ {
    list.Append(rand.Intn(23))
  }

  types.PrintLinkedList(list)
  kToLast2(list.Head, 1)
}
